zerver/models.py

```diff
@@ -1806,6 +1803,11 @@ def can_edit_topic_of_any_message(self) -> bool:
             return True
         return self.has_permission("edit_topic_policy")
 
+    def can_delete_own_message(self) -> bool:
+        if self.realm.delete_own_message_policy == Realm.POLICY_EVERYONE:
+            return True
```

Comment:
I think `POLICY_EVERYONE` should be just done inside `has_permission`, since we may use for other endpoints in the future as well, just like we do for `POLICY_NOBODY`.